### DISCUSSION TITLE
Suggeting using AUTOMATIC1111 Extensions API instead of forking it

### DISCUSSION THREAD

**Discussion by a-l-e-x-d-s-9** [Interpause/auto-sd-krita#46] [Ideas]
I have noticed that AUTOMATIC1111 started separating things like Image Browser to the Extensions folder, with a separate git repository for each extension. It seems like a better software design for this plugin too, to keep a better separation of responsibilities, and modular encapsulation of the code. As an extension, you would not be forced to chase after AUTOMATIC1111 daily. Possibly some additions to AUTOMATIC1111 Extensions API would be necessary, but once negotiated and agreed upon, you will have much less noise from AUTOMATIC1111.
Please consider this option. I think it is a better solution than forking AUTOMATIC1111.

**Comment by Interpause** [comment]
I will look at it after im back from my break, but I will be busy for quite a while so I might not have time for a major change like this. There is also the option of waiting for the upstream official API to be complete and mature.

**Reply by Interpause** [reply]
wait ive 1 hour now, that might be enough time:
https://github.com/Interpause/auto-sd-paint-ext

**Comment by Interpause** [comment]
https://github.com/Interpause/auto-sd-paint-ext
It doesn't work yet, but looks like the change should not be as hard as I thought.

Unfortunately, my forced staycation has begun so I will be pausing work on this till Sunday night.
